Maple2.Database/Storage/Game/GameStorage.Club.cs

```diff
@@ -1,64 +1,148 @@
+            int count = Context.ClubMember.Where(member => member.ClubId == clubId && member.CharacterId == characterId).Delete();
+            return SaveChanges() && count > 0;
+        }
 
-            Context.Club.Add(model);
+        public List<ClubMember> GetClubMembers(IPlayerInfoProvider provider, long clubId) {
+            return Context.ClubMember.Where(member => member.ClubId == clubId)
+                .AsEnumerable()
+                .Select(member => {
+                    PlayerInfo? info = provider.GetPlayerInfo(member.CharacterId);
+                    return info == null ? null : new ClubMember {
+                        Info = info,
+                        JoinTime = member.CreationTime.ToEpochSeconds(),
+                        ClubId = member.ClubId,
+                        LoginTime = member.LoginTime.ToEpochSeconds(),
+                    };
+                }).WhereNotNull().ToList();
+        }
 
-            // I know this is an extra read, but the conversion logic is complicated.
-            return Context.TrySaveChanges() ? GetClub(model.Id) : null;
+        public bool SaveClub(Club club) {
+            BeginTransaction();
+            if (!Context.Club.Any(model => model.Id == club.Id)) {
+                return false;
+            }
+            Context.Club.Update(club);
+            SaveClubMembers(club.Id, club.Members.Values);
+            if (!SaveChanges()) {
+                return false;
+            }
+            return Commit();
         }
 
-        public IList<ClubMember> GetClubMembers(long clubId) {
-            var results = (from member in Context.ClubMember where member.ClubId == clubId
-                           join account in Context.Account on member.Character.AccountId equals account.Id
-                           join indoor in Context.UgcMap on
-                               new { OwnerId = member.Character.AccountId, Indoor = true } equals new { indoor.OwnerId, indoor.Indoor }
-                           join outdoor in Context.UgcMap on
-                               new { OwnerId = member.Character.AccountId, Indoor = true } equals new { outdoor.OwnerId, outdoor.Indoor } into plot
-                           from outdoor in plot.DefaultIfEmpty()
-                           select new { account, member, indoor, outdoor })
-                .AsEnumerable();
-
-            return results.Select(result => {
-                AchievementInfo achievementInfo = GetAchievementInfo(result.account.Id, result.member.CharacterId);
-                PlayerInfo playerInfo = BuildPlayerInfo(result.member.Character, result.indoor, result.outdoor, achievementInfo, result.account.PremiumTime);
-                return new ClubMember(playerInfo, result.member.CreationTime.ToEpochSeconds(), result.member.Character.LastModified.ToEpochSeconds());
-            }).ToList();
+        public bool SaveClubMembers(long clubId, ICollection<ClubMember> members) {
```

Comment:
I would honestly just merge this with `SaveClub` above. If you don't want to do that there are a couple of things you should clean up:
1. return value is unused
2. it doesn't actually save anything, it just updates the context (so rename accordingly, or perform a save)
3. this is never called elsewhere so make private